src/javascript/app/common/content_visibility.js

```diff
@@ -42,22 +41,27 @@ const updateTabDisplay = require('../../_common/tab_selector').updateTabDisplay;
             data-show='Costarica'         -> throws error
 */
 
-const visible_classname = 'data-show-visible';
-const mt_company_rule   = 'mtcompany';
+const visible_classname    = 'data-show-visible';
+const mt_company_rule      = 'mtcompany';
+const mt_country_shortcode = 'mt';
 
 const ContentVisibility = (() => {
     const init = () => {
-        if (Client.isLoggedIn()) {
-            BinarySocket.wait('authorize', 'landing_company').then(() => {
-                controlVisibility(
-                    State.getResponse('authorize.landing_company_name'),
-                    MetaTrader.isEligible(),
-                    State.getResponse('landing_company.mt_financial_company.shortcode')
-                );
+        BinarySocket.wait('authorize', 'landing_company', 'website_status').then(() => {
+            const is_virtual                = Client.isLoggedIn() && Client.get('is_virtual');
+            const clients_country           = State.getResponse('website_status.clients_country');
+            const client_is_mt_country      = clients_country === mt_country_shortcode;
+            const landing_company_shortcode = is_virtual ?
+                (State.getResponse('landing_company.financial_company.shortcode') ||
+                State.getResponse('landing_company.gaming_company.shortcode') ||
+                (client_is_mt_country ? 'malta' : 'default')) :
+                Client.get('landing_company_shortcode') || (client_is_mt_country ? 'malta' : 'default');
```

Comment:
we shouldn't check `client_is_mt_country ? 'malta' : 'default'` for logged in client at all. logged in client should be decided based on residence alone, and not IP address.

Reply:
This still doesn't work for logged out EU IP address. is_virtual is false, so it tries to fetch landing_company_shortcode, which is not set for logged out clients, and if country is anything other than malta, landing company shortcode becomes default so it shows the non-EU content. for logged out clients we need to be behaving the same as how virtual client is retrieving the shortcode, not how real accounts should be retrieving shortcode.